[RF] New RooFit batch mode with GPU support (#9004)
root-project/root: roofit/batchcompute/README.md

If `RooBatchCompute::Cuda` is selected, RooFit will launch CUDA kernels for computing possibilities and potentially other intense computations. At the same time, the most efficent cpu library loaded will also handle parts of the computations in parallel with the GPU (or potentially, if it's faster all of them), thus gaining full advantage of the available hardware. For this purpose `RooFitDriver`, a newly created RooFit class (in roofitcore) takes over the task of analyzing the computations and assigning each to the correct piece of hardware, taking into consideration the performance boost or penalty that may arise with every method of computing.

#### Multithread computations
The CPU instance of the computing library can furthermore execute multithread computations. This also applies for computations handled by the CPU in the `RooBatchCompute::Cuda` mode. To use them, one needs to set the desired number of parallel tasks before calling `fitTo()` as shown below:
```c++
  ROOT::EnableImplicitMT(nThreads);
  RooMyPDF.fitTo(data, BatchMode(RooBatchCompute::Cpu)); // can also use RooBatchCompute::Cuda
```

> ### User made PDFs

Reviewer: ```suggestion
### User-made PDFs
```

Author: Ok, fixed!

The easiest and most efficient way of accelerating your PDFs is to request their addition to the official RooFit by submiting a ticket [here](https://github.com/root-project/root/issues/new). The ROOT team will gladly assist you and take care of the details.

> The above process might take some time and the users will be required to update ROOT to use the newly introduced PDFs. In the meantime, you are able to significantly improve the speed of fitting (but not take full advantage of the RooBatchCompute library), at least by using the batch evaluation feature.

Reviewer: ```suggestion
While your code is integrated, you are able to significantly improve the speed of fitting (but not take full advantage of the RooBatchCompute library), at least by using the batch evaluation feature.
```

Author: Thanks, I updated this!
